rust/worker/src/server.rs

```diff
@@ -806,4 +733,199 @@ mod tests {
+        let err = response.unwrap_err();
+        assert_eq!(err.code(), tonic::Code::InvalidArgument);
+        assert!(err.message().contains("context"));
+
+        // invalid vector
+        let mut request = first_request.clone();
+        request.vectors = vec![Vector {
+            dimension: 1,
+            vector: vec![0],
+            encoding: 0,
+        }];
+
+        let response = reader.query_vectors(request).await;
+        assert!(response.is_err());
+        let err = response.unwrap_err();
+        assert_eq!(err.code(), tonic::Code::Internal);
+        let sub_str = "converting vector";
+        assert!(
+            err.message().contains(sub_str),
+            "message: {}, sub_str: {}",
+            err.message(),
+            sub_str
+        );
+    }
+
+    #[tokio::test]
+    #[cfg(debug_assertions)]
+    async fn validate_query_metadata_request() {
+        use chroma_proto::metadata_reader_client::MetadataReaderClient as Client;
+        use chroma_types::chroma_proto::QueryMetadataRequest as Request;
+
+        let mut reader = Client::connect(run_server()).await.unwrap();
+
+        let first_request = Request {
+            collection_id: COLLECTION_UUID.to_string(),
+            segment_id: SEGMENT_UUID.into(),
+            version_context: Some(RequestVersionContext {
+                collection_version: 0,
+                log_position: 0,
+            }),
+            ..Default::default()
+        };
+
+        // segment or collection
+        let response = reader.query_metadata(first_request.clone()).await;
+        assert!(response.is_err());
+        assert_eq!(response.unwrap_err().code(), tonic::Code::NotFound);
+
+        // invalid collection uuid
+        let mut request = first_request.clone();
+        request.collection_id = INVALID_UUID.into();
+        let response = reader.query_metadata(request).await;
+
+        assert!(response.is_err());
+        let err = response.unwrap_err();
+        assert_eq!(err.code(), tonic::Code::InvalidArgument);
+        assert!(err.message().contains("Collection UUID"));
+
+        // invalid segment uuid
+        let mut request = first_request.clone();
+        request.segment_id = INVALID_UUID.into();
+        let response = reader.query_metadata(request).await;
+
+        assert!(response.is_err());
+        let err = response.unwrap_err();
+        assert_eq!(err.code(), tonic::Code::InvalidArgument);
+        assert!(err.message().contains("Segment UUID"));
+
+        // invalid version context
+        let mut request = first_request.clone();
+        request.version_context = None;
+        let response = reader.query_metadata(request).await;
+
+        assert!(response.is_err());
+        let err = response.unwrap_err();
+        assert_eq!(err.code(), tonic::Code::InvalidArgument);
+        assert!(err.message().contains("context"));
+    }
+
+    #[cfg(debug_assertions)]
+    fn to_byte_slice<T>(v: &[T]) -> &[u8] {
```

Comment:
Would prefer this was just for f32 to minimize accidental usage in the wrong way.

Reply:
I made this function generic to allow for testing with different encodings and to validate each one. Since it is defined within the tests modules, it's only visible during testing. However, I'm open to changing it to accept f32 directly until a more generic version is necessary. Let me know if you prefer that approach!

Reply:
I agree 🙂; we can always make the function generic later if needed. I’ve updated the code to accept f32 as suggested.

Reply:
I agree with @HammadB; we can always make the function generic later if needed. I’ve updated the code to accept f32 as suggested.

Reply:
Cool yeah I just prefer we don’t prematurely expand unsafe code, even it tests.

thanks for this change by the way!

Reply:
Make sense to me 🙂